src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayPipelineCommand.java

```diff
@@ -67,7 +67,7 @@
             // which currently has no protected method allowing getItemByFullName to be replaced.
             throw new AbortException("Not a Pipeline build");
         }
-        if (!action.isEnabled()) {
+        if (!action.isEnabled() || !action.isReplayableSandboxTest()) {
```

Comment:
BTW I would suggest marking `isEnabled` restricted just to be sure no unknown party is innocently calling it and thinking it secure.